corehq/apps/cleanup/utils.py

```diff
@@ -67,3 +80,55 @@ def migrate_to_deleted_on(db_cls, old_field, should_audit=False):
         update_kwargs['audit_action'] = AuditAction.AUDIT
     update_count = queryset.update(**update_kwargs)
     return update_count
+
+
+def hard_delete_sql_objects_before_cutoff(cutoff):
+    """
+    Permanently deletes SQL objects with deleted_on set to a datetime earlier
+    than the specified cutoff datetime
+    :param cutoff: datetime used to obtain objects to be hard deleted
+    :return: dictionary of count of deleted objects per table
+    """
+    counts = {}
+    tables = [AutomaticUpdateRule, AlertSchedule, TimedSchedule, ScheduledBroadcast, ImmediateBroadcast]
+    audited_tables = [AutomaticUpdateRule]
+    for table in tables:
+        kwargs = {}
+        if table in audited_tables:
+            kwargs = {'audit_action': AuditAction.AUDIT}
+
+        queryset = table.objects.filter(deleted_on__lt=cutoff)
+        if queryset.exists():
+            counts[table.__name__] = queryset.count()
```

Comment:
you don't need to do a count since `delete` will return the counts: https://docs.djangoproject.com/en/4.2/ref/models/querysets/#delete

Reply:
🤦🏻‍♂️

Reply:
See cc1c49520080070bb29bb593464998a303e8cfc2